osf_tests/test_notable_domains.py

```diff
@@ -115,3 +116,204 @@ def test_spam_check(self, app, factory, spam_domain, marked_as_spam_domain, requ
+        )
+
+    @pytest.fixture()
+    def ignored_notable_domain(self, ignored_domain):
+        return NotableDomain.objects.create(
+            domain=ignored_domain.netloc,
+            note=NotableDomain.Note.IGNORED,
+        )
+
+    @pytest.mark.parametrize('factory', [NodeFactory, CommentFactory, PreprintFactory, RegistrationFactory])
+    def test_from_spam_to_unknown(self, factory, spam_domain_one, spam_domain_two, unknown_domain, ignored_domain, spam_notable_domain_one, spam_notable_domain_two, unknown_notable_domain, ignored_notable_domain):
+        obj_one = factory()
+        obj_two = factory()
+        obj_three = factory()
+        check_resource_for_domains.apply_async(
+            kwargs=dict(
+                guid=obj_one.guids.first()._id,
+                content=f'{spam_domain_one.geturl()} {unknown_domain.geturl()} {ignored_domain.geturl()}',
+            )
+        )
+        check_resource_for_domains.apply_async(
+            kwargs=dict(
+                guid=obj_two.guids.first()._id,
+                content=f'{spam_domain_one.geturl()} {spam_domain_two.geturl()} {unknown_domain.geturl()} {ignored_domain.geturl()}',
+            )
+        )
+        obj_three.spam_data['who_flagged'] = 'some external spam checker'
+        obj_three.save()
+        check_resource_for_domains.apply_async(
+            kwargs=dict(
+                guid=obj_three.guids.first()._id,
+                content=f'{spam_domain_one.geturl()} {unknown_domain.geturl()} {ignored_domain.geturl()}',
+            )
+        )
+        obj_one.reload()
+        obj_two.reload()
+        obj_three.reload()
+        assert obj_one.spam_status == SpamStatus.SPAM
+        assert obj_two.spam_status == SpamStatus.SPAM
+        assert obj_three.spam_status == SpamStatus.SPAM
+        assert set(obj_one.spam_data['domains']) == set([spam_domain_one.netloc])
+        assert set(obj_two.spam_data['domains']) == set([spam_domain_one.netloc, spam_domain_two.netloc])
+        assert set(obj_three.spam_data['domains']) == set([spam_domain_one.netloc])
+        spam_notable_domain_one.note = NotableDomain.Note.UNKNOWN
+        spam_notable_domain_one.save()
+        obj_one.reload()
+        obj_two.reload()
+        obj_three.reload()
+        assert obj_one.spam_status == SpamStatus.UNKNOWN
+        assert obj_two.spam_status == SpamStatus.SPAM
+        assert obj_three.spam_status == SpamStatus.SPAM
+        assert len(obj_one.spam_data['domains']) == 0
+        assert set(obj_two.spam_data['domains']) == set([spam_domain_two.netloc])
+        assert len(obj_three.spam_data['domains']) == 0
+
+    @pytest.mark.parametrize('factory', [NodeFactory, CommentFactory, PreprintFactory, RegistrationFactory])
+    def test_from_spam_to_ignored(self, factory, spam_domain_one, spam_domain_two, unknown_domain, ignored_domain, spam_notable_domain_one, spam_notable_domain_two, unknown_notable_domain, ignored_notable_domain):
```

Comment:
nit: this is really three tests, if this fails, depending on where it fails, we don't know if just one case or multiple cases are broken.

You could parameterize it further with something like
```
@pytest.mark.parametrize(
    'content_domains, flagged_externally, expected_spam_state',
   [
        ([spam_domain_one, unknown_domain, flagged_domain], False, SPAM_STATUS.UNKNOWN),
        ([spam_domain_one, spam_domain_two], False, SPAM_STATUS.SPAM),
        ([spam_domain_one], True, SPAM_STATUS.SPAM)
    ]
```

(note that the domains themselves can be module-level constants instead of fixtures)